### PR TITLE
Velocity distribution 

### PR DESCRIPTION
**Pull request by aguspesce**
Create a linear velocity distribution in the lateral boundaries where the velocity is zero for z > z_start and for z_min < z < z_start assume a linear increase of velocity until the bottom of the model.
Also, create a function to save the velocity distribution in the MD3D/MD2D  format

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to the `examples`.
- [ ] If new dependency is required, add it to `environment.yml`, `requirements.txt`, `setup.py` and `requirements-dev.txt` (if it's a development dependency).

### CODE REVIEW

File: modelling_earth/velocity.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        )
+    # Define the shape of the coordinates
+    if dimension == 3:
+        shape = (
+            coordinates["x"].shape[0],
+            coordinates["y"].shape[0],
+            coordinates["z"].shape[0],
+        )
+    if dimension == 2:
+        shape = (coordinates["x"].shape[0], coordinates["z"].shape[0])
```
Comment (santisoler) [inline modelling_earth/velocity.py]: These two if statements that have the goal to define the shape can be summarized on a single line:

```python
shape = tuple(coordinates[i].size for i in expected_dims)
```

File: modelling_earth/velocity.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    # Initialize the velocity dataset with zero values
+    velocity = xr.Dataset(coords=coordinates)
+    velocity["velocity_x"] = (coordinates, np.zeros(shape))
+    velocity["velocity_z"] = (coordinates, np.zeros(shape))
+    if dimension == 3:
+        velocity["velocity_y"] = (coordinates, np.zeros(shape))
```
Comment (santisoler) [inline modelling_earth/velocity.py]: Same here:

```python
for i in expected_dims:
    velocity["velocity_{}".format(i)] = (coordinates, np.zeros(shape))
```

File: modelling_earth/velocity.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    # Calculate the velocity assume a linear increase with depth and add increasing
+    # velocity values on the first and last column using a condition
```
Comment (santisoler) [inline modelling_earth/velocity.py]: Fix the writing of this comment. Maybe:
```
Calculate the velocity values.
Assume a linear increase with depth ...
These velocity values must be added on the first and last column.
```

File: modelling_earth/velocity.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    Parameters
+    ----------
+    coordinates : :class:`xarray.DataArrayCoordinates`
+        Coordinates located on a regular grid where the temperature distribution will be
+        created. Must be in meters and can be either 2D or 3D.
+    z_start : float
+        Depth value where the velocity condition changes in meters.
+    velocity_bottom :
+        Velocity at bottom boundary (z_min) on the x and z axes in m/ in m/s.
```
[the remark this answers is not in the document]
Comment (aguspesce) [reply modelling_earth/velocity.py]: done

File: modelling_earth/velocity.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+"""
+Functions to create 2D and 3D velocity distributions
+"""
+
+import numpy as np
+import xarray as xr
+
+
+def linear_velocity(coordinates, z_start, velocity_bottom=(0, 0, 0)):
+    """
```
[the remark this answers is not in the document]
Comment (aguspesce) [reply modelling_earth/velocity.py]: Done

File: modelling_earth/velocity.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    Create a linear velocity distribution in the lateral boundaries where the velocity
+    is zero for z > z_start and for z_min < z < z_start assume a linear increase of
```
[the remark this answers is not in the document]
Comment (aguspesce) [reply modelling_earth/velocity.py]: I try to do it. See the new commit and tell me your opinion. :)

File: modelling_earth/velocity.py (hunk 1 above)
Comment (aguspesce) [reply modelling_earth/velocity.py]: Done

File: modelling_earth/velocity.py (hunk 2 above)
Comment (aguspesce) [reply modelling_earth/velocity.py]: Done! Thank you for your suggestion.

File: modelling_earth/velocity.py (hunk 3 above)
Comment (aguspesce) [reply modelling_earth/velocity.py]: Done

File: modelling_earth/velocity.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+Functions to create 2D and 3D velocity distributions
+"""
+
+import numpy as np
+import xarray as xr
+
+
+def linear_velocity(coordinates, z_start, velocity_bottom, direction="x"):
+    """
+    Hotizontal velocity distribution along a define direction.
```
Comment (santisoler) [inline modelling_earth/velocity.py]:
```suggestion
    Hotizontal velocity distribution along a define direction.
    
```

File: modelling_earth/velocity.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        velocity["velocity_{}".format(i)] = (coordinates, np.zeros(shape))
+
+    # Calculate the velocity.
+    # Assume a linear increase with depth where where the velocity is 0 for
+    # coordines.z > z_start and for coordinates.z[0] < z < z_start.
+    # These velocity values must be added on the first and last column.
+    # Define the condition to calculate the velocity
+    condition = np.logical_and(
+        velocity.z < z_start,
+        np.logical_or(velocity.x == velocity.x[0], velocity.x == velocity.x[-1]),
```
Comment (santisoler) [inline modelling_earth/velocity.py]: You forgot to change `x` with the passed `direction`. Remember you can use `getattr` to get an attribute of a class by a string variable. So for example, in this case you can use `getattr(velocity, direction)` to get the desired coordinate.

File: modelling_earth/velocity.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    )
+    # Calculate the linear increase
+    linear_increase = _linear_velocity_calculation(velocity, z_start, velocity_bottom)
+    # Add the linear increase along thr direction if subduction acording the condition
```
Comment (santisoler) [inline modelling_earth/velocity.py]:
```suggestion
    # Add the linear increase along the direction if subduction according the condition
```

File: modelling_earth/velocity.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    velocity["velocity_{}".format(direction)] = velocity[
+        "velocity_{}".format(direction)
+    ].where(~condition, linear_increase)
+    return velocity
+
+
+def _check_velocity(coordinates, velocity_value):
+    """
+    Check if coordinates is 2D or 3D.
```
Comment (santisoler) [inline modelling_earth/velocity.py]:
```suggestion
    Check if coordinates are 2D or 3D.
```

File: modelling_earth/io/velocity.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        Path to save the velocity file.
+    fname : str (optional)
+       Filename of the output ASCII file. Default to ``veloc_0_3D.txt``.
+    """
+    dimension = len(velocity.dims)
+    # Initialize the velocity to save
+    shape = velocity.velocity_x.shape
+    if dimension == 3:
+        velocity_to_save = np.zeros((dimension, shape[0] * shape[1] * shape[2]))
+        # Revel the velocity component and put them in only file
```
Comment (santisoler) [inline modelling_earth/io/velocity.py]:
```suggestion
        # Ravel the velocity components and save them to a single file
```

File: modelling_earth/velocity.py (hunk 8 above)
Comment (aguspesce) [reply modelling_earth/velocity.py]: Done!

Comment (aguspesce) [reply modelling_earth/velocity.py]: Done! Thank you for the data

**Pull request merged**